backend/api.test/EventHandlers/TestMissionEventHandler.cs

```diff
@@ -198,6 +175,7 @@ public async void NewMissionIsStartedWhenRobotBecomesAvailable()
 
             // Act
             _mqttService.RaiseEvent(nameof(MqttService.MqttIsarRobotStatusReceived), mqttEventArgs);
+            Thread.Sleep(500);
```

Comment:
What happens without the sleeps, how did they work before without them and how did you settle on the specific timings (since some of them differ)?

Reply:
I will honestly say I'm not sure how they functioned before. My suspicion is that the mocking we had in place for the robot controller meant the tests were so quick that they managed to reliably perform the operations before the assert was called. After removing the mocks it was very clear that there is a timing issue which is not surprising giving the design of the system. 

The specific times are not scientific but rather taken as run the test a few times, see what is suitable but not so high that it impacts test runtime too much. I think sleeps here are okey as the tests aren't very complicated. It's mainly "nice the other thread some time".